ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -1,13 +1,14 @@
 import classNames from 'classnames';
-import * as React from 'react';
-import * as ReactDOM from 'react-dom';
+import React, {ComponentType, ReactNode, useEffect, useRef} from 'react';
+import {useState} from 'react';
+import {createPortal} from 'react-dom';
 import {BehaviorSubject, fromEvent, merge, Subscription} from 'rxjs';
 import {filter} from 'rxjs/operators';
 
 export interface DropDownProps {
     isMenu?: boolean;
-    anchor: React.ComponentType;
-    children: React.ReactNode | (() => React.ReactNode);
```

Comment:
ah I see that the function form is never actually this component's [sole consumer](https://github.com/argoproj/argo-workflows/blob/e91c2737f3dff1fee41ce97991e294a57c53fc93/ui/src/app/shared/components/filter-drop-down.tsx#L18) -- good simplification!